dispatcher_plugin_nb2workflow/dataserver_dispatcher.py

```diff
@@ -257,3 +263,36 @@ def run_query(self,
                                  extra_message = res.text)
 
         return res, query_out
+
+    def update_param_dict_download_file_url(self, param_dict, task, session_id, job_id, instrument_name, token, basepath):
+        for param in param_dict:
+            param_obj = self.backend_options[task]['parameters'].get(param, None)
+            # TODO improve this check, is it enough?
+            if param_obj is not None \
+                    and param_obj.get('owl_type') == "http://odahub.io/ontology#POSIXPath" \
```

Comment:
It may be a subclass in the future (e.g. specifying file type). Also, if there are other annotations, the `owl_type` will be constructed by nb2workflow and the relation with POSIXPath will be in the `extra_ttl`. 
So that for this check [`Ontology.get_parameter_hierarchy()`](https://github.com/oda-hub/oda_api/blob/aa2e190cbe9deb6fcd360f7953962791f91c200d/oda_api/ontology_helper.py#L244) needs to be used

Reply:
But creating the ontology object here and parsing the parameter object again (already did outside) will introduce an overhead (it's rather expensive). 
So it will be better to either patch the parameter value outside (e.g. in Parameter object, modifying `default_value()`) or somehow pass this info here (both require additional changes in the main dispatcher code)

Reply:
> So that for this check [`Ontology.get_parameter_hierarchy()`](https://github.com/oda-hub/oda_api/blob/aa2e190cbe9deb6fcd360f7953962791f91c200d/oda_api/ontology_helper.py#L244) needs to be used

Is there an example I can refer to ?

Reply:
> Is there an example I can refer to ?

An example is how it's implemented in `Parameter.from_owl_uri()`. And before being sent into `run_query` by its value, the parameter was already created with this method. So this info is already around, the problem is that it's not available in the scope of `NB2WDataDispatcher`

Reply:
Overall, this seems to be some change to be implemented.

> So it will be better to either patch the parameter value outside (e.g. in Parameter object, modifying `default_value()`) or somehow pass this info here (both require additional changes in the main dispatcher code)

Probably the first options is the most feasible

Reply:
the implementation now relies on the `get_default_value` method that. in case of FileReference objects returns the information that the file can be downloaded, so the `download_file` url can be created